proto/config/config.proto

```diff
@@ -0,0 +1,89 @@
+syntax = "proto3";
+
+package linkerd2.config;
+
+option go_package = "github.com/linkerd/linkerd2/controller/gen/config";
+
+// ConfigStore provides an abstraction over the underlying configuration storge.
+service ConfigStore {
+  // Return the global config.
+  rpc GetGlobalConfig(GetParam) returns (GlobalConfig) {}
+
+  // Return the proxy config.
+  rpc GetProxyConfig(GetParam) returns (ProxyConfig) {}
+
+  // Set the global config. It overwrites the existing value.
+  rpc SetGlobalConfig(GlobalConfig) returns (Result) {}
+
+  // Set the proxy config. It overwrites the existing value.
+  rpc SetProxyConfig(ProxyConfig) returns (Result) {}
+}
+
+message GlobalConfig {
+  string linkerd_namespace = 1;
+  bool cni_enabled = 2;
+  bool tls_enabled = 3;
+}
+
+message ProxyConfig {
+  Image proxy_image = 1;
+  Image proxy_init_image = 2;
+
+  Port api_port = 3;
+  Port control_port = 4;
+  repeated Port ignore_inbound_ports = 5;
+  repeated Port ignore_outbound_ports = 6;
+  Port inbound_port = 7;
+  Port metrics_port = 8;
+  Port outbound_port = 9;
+
+  ResourceRequirements resource = 10;
+
+  TLS tls_enabled = 11;
+
+  int64 proxy_uid = 12;
+  LogLevel log_level = 13;
+  bool disable_external_profiles = 14;
+}
+
+message Image {
+  string image_name = 1;
+  string pull_policy = 2;
+  string registry = 3;
+}
+
+message Port {
+  uint32 port = 1;
+}
+
+message ResourceRequirements {
+  string request_cpu = 1;
+  string request_memory = 2;
+}
+
+message LogLevel {
+  string level = 1;
+}
+
+enum TLS {
+  NONE = 0;
+  OPTIONAL = 1;
+}
```

Comment:
Let's avoid introducing this type if we can. As I mentioned above, this is going to chagne substantially in the 2.3 release.